src/jit/codegenxarch.cpp

```diff
@@ -1329,9 +1345,23 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
     }
     else
     {
-        // dividend must be in RAX
-        if (dividend->gtRegNum != REG_RAX)
+        if (dividendIsLong)
+        {
+            assert(dividendLo != nullptr && dividendHi != nullptr);
+
+            // dividendLo must be in RAX; dividendHi must be in RDX
+            if (dividendLo->gtRegNum != REG_RAX)
```

Comment:
Might be worth changing the code that inserts these `MOV`s into asserts (or at least filing an issue to do so and linking it to the code).